pymc_extras/statespace/models/structural/components/measurement_error.py

```diff
@@ -122,25 +128,36 @@ def __init__(
             share_states=share_states,
         )
 
-    def populate_component_properties(self):
+    def _set_parameters(self) -> None:
         k_endog = self.k_endog
         k_endog_effective = 1 if self.share_states else k_endog
 
-        self.param_names = [f"sigma_{self.name}"]
-        self.param_dims = {}
-        self.coords = {}
+        sigma_param = Parameter(
+            name=f"sigma_{self.name}",
+            shape=(k_endog_effective,) if k_endog_effective > 1 else (),
+            dims=(f"endog_{self.name}",) if k_endog_effective > 1 else None,
+            constraints="Positive",
+        )
+
+        self.param_info = ParameterInfo(parameters=[sigma_param])
+
+    def _set_coords(self) -> None:
+        k_endog = self.k_endog
+        k_endog_effective = 1 if self.share_states else k_endog
+        observed_state_names = self.observed_state_names
+
+        coords_container = []
 
         if k_endog_effective > 1:
-            self.param_dims[f"sigma_{self.name}"] = (f"endog_{self.name}",)
-            self.coords[f"endog_{self.name}"] = self.observed_state_names
-
-        self.param_info = {
-            f"sigma_{self.name}": {
-                "shape": (k_endog_effective,) if k_endog_effective > 1 else (),
-                "constraints": "Positive",
-                "dims": (f"endog_{self.name}",) if k_endog_effective > 1 else None,
-            }
-        }
+            endog_coord = Coord(dimension=f"endog_{self.name}", labels=observed_state_names)
+            coords_container.append(endog_coord)
+
+        self.coords_info = CoordInfo(coords=coords_container)
+
+    def populate_component_properties(self) -> None:
```

Comment:
You should implement all the other mandatory setters too (the ones that raise `NotImplementedError` by default). Just have them do nothing or set an empty `Info`. Not sure which is easier downstream. 

If we always set an empty Info, the logic for joining components becomes a bit easier, because we never have to worry about the `None` case. If its a lot of work no need to worry about it in this PR.